packages/celotool/src/e2e-tests/utils.ts

```diff
@@ -385,109 +125,155 @@ export async function migrateContracts(
+  } catch (error) {
+    const revertFound = error.message.search('revert') >= 0
+    if (errorMessage === '') {
+      assert(revertFound, `Expected "revert", got ${error} instead`)
+    } else {
+      assert(revertFound, errorMessage)
+    }
   }
-  return startGeth(gethBinaryPath, instance)
 }
 
-// Add validator 0 as a peer of each other validator.
-async function connectValidatorPeers(gethConfig: GethTestConfig) {
-  const admins = gethConfig.instances
-    .filter(({ wsport, rpcport, validating }) => validating && (wsport || rpcport))
-    .map(
-      ({ wsport, rpcport }) =>
-        new Admin(`${wsport ? 'ws' : 'http'}://localhost:${wsport || rpcport}`)
-    )
-  const enodes = await Promise.all(admins.map(async (admin) => (await admin.getNodeInfo()).enode))
-  await Promise.all(
-    admins.map(async (admin, i) => {
-      await Promise.all(
-        enodes.map(async (enode, j) => {
-          if (i === j) {
-            return
-          }
-          await admin.addPeer(enode)
-        })
-      )
-    })
-  )
+export function getHooks(gethConfig: GethRunConfig) {
+  return getContext(gethConfig).hooks
 }
 
-export function getContext(gethConfig: GethTestConfig) {
+export function getContext(gethConfig: GethRunConfig, verbose: boolean = verboseOutput) {
   const mnemonic =
     'jazz ripple brown cloth door bridge pen danger deer thumb cable prepare negative library vast'
   const validatorInstances = gethConfig.instances.filter((x: any) => x.validating)
+
   const numValidators = validatorInstances.length
+
   const validatorPrivateKeys = getPrivateKeysFor(AccountType.VALIDATOR, mnemonic, numValidators)
   const attestationKeys = getPrivateKeysFor(AccountType.ATTESTATION, mnemonic, numValidators)
-  const validators = getValidators(mnemonic, numValidators)
+  const validators = getValidatorsInformation(mnemonic, numValidators)
+
+  const proxyInstances = gethConfig.instances.filter((x: any) => x.isProxy)
+  const numProxies = proxyInstances.length
+
+  const proxyPrivateKeys = getPrivateKeysFor(AccountType.PROXY, mnemonic, numProxies)
+  const proxyEnodes = proxyPrivateKeys.map((x: any, i: number) => [
+    proxyInstances[i].name,
+    getEnodeAddress(privateKeyToPublicKey(x), '127.0.0.1', proxyInstances[i].proxyport!),
+    getEnodeAddress(privateKeyToPublicKey(x), '127.0.0.1', proxyInstances[i].port),
+  ])
+
   const argv = require('minimist')(process.argv.slice(2))
   const branch = argv.branch || 'master'
-  const gethRepoPath = argv.localgeth || '/tmp/geth'
-  const gethBinaryPath = `${gethRepoPath}/build/bin/geth`
+
+  gethConfig.gethRepoPath = argv.localgeth || '/tmp/geth'
+  const gethBinaryPath = `${gethConfig.gethRepoPath}/build/bin/geth`
+  const bootnodeBinaryPath = `${gethConfig.gethRepoPath}/build/bin/bootnode`
 
   const before = async () => {
-    if (!argv.localgeth) {
-      await checkoutGethRepo(branch, gethRepoPath)
+    if (!argv.localgeth && gethConfig.gethRepoPath) {
```

Comment:
`gethConfig.gethRepoPath` should always be truthy based on the code above.
```suggestion
    if (!argv.localgeth) {
```